src/containers/ProfilePage/ProfilePage.js

```diff
@@ -142,6 +151,50 @@ export const DesktopReviews = props => {
   );
 };
 
+export const CustomUserFields = props => {
+  const { publicData, metadata, userFields } = props;
+  return (
+    <>
+      <H4 as="h2" className={css.listingsTitle}>
+        <FormattedMessage id="ProfilePage.customFieldsHeading" />
+      </H4>
+      <SectionDetailsMaybe {...props} />
+      {userFields.reduce((pickedElements, config) => {
+        const { key, enumOptions, includeForListingTypes, scope = 'public' } = config;
+        // TODO fix isTargetListingType => isTargetUserType
+        const listingType = publicData?.listingType;
+        const isTargetListingType =
+          includeForListingTypes == null || includeForListingTypes.includes(listingType);
+
+        const createFilterOptions = options =>
+          options.map(o => ({ key: `${o.option}`, label: o.label }));
+
+        const value =
+          scope === 'public' && !!publicData
+            ? publicData[key]
+            : scope === 'metadata' && !!metadata
+            ? metadata[key]
+            : null;
+
+        const hasValue = value != null;
+        return isTargetListingType && config.schemaType === SCHEMA_TYPE_MULTI_ENUM
+          ? [
+              ...pickedElements,
+              <SectionMultiEnumMaybe
+                key={key}
+                heading={config?.label}
+                options={createFilterOptions(enumOptions)}
+                selectedOptions={value || []}
+              />,
+            ]
+          : isTargetListingType && hasValue && config.schemaType === SCHEMA_TYPE_TEXT
+          ? [...pickedElements, <SectionTextMaybe key={key} heading={config?.label} text={value} />]
+          : pickedElements;
+      }, [])}
```

Comment:
Perhaps we could extract the props generation into a new function (added to some util file)

```js
const pickCustomFieldProps = (publicData, metadata, entityTypeKey, activeEntityTypes=[]) => {
  return userFields.reduce((pickedElementProps, config) => {
        const { key, enumOptions, includeForListingTypes, scope = 'public', schemaType } = config;
        // TODO fix isTargetListingType => isTargetUserType
        const entityType = publicData?.[entityTypeKey];
        const isTargetEntityType =
          activeEntityTypes == null || activeEntityTypes.includes(entityType);

        const createFilterOptions = options =>
          options.map(o => ({ key: `${o.option}`, label: o.label }));

        const value =
          scope === 'public' && !!publicData
            ? publicData[key]
            : scope === 'metadata' && !!metadata
            ? metadata[key]
            : null;

        const hasValue = value != null;
        return isTargetEntityType && schemaType === SCHEMA_TYPE_MULTI_ENUM
          ? [
              ...pickedElementProps,
              {
                schemaType,
                key,
                heading: config?.label,
                options: createFilterOptions(enumOptions),
                selectedOptions: value || [],
              },
            ]
          : isTargetEntityType && hasValue && config.schemaType === SCHEMA_TYPE_TEXT
          ? [...pickedElementProps, { schemaType, key heading: config?.label, text: value }]
          : pickedElementProps;
      }, []);
```

Then we could use that here and on ListingPage(s)

```js
// before JSX return statement: 
// const propsForCustomFields = pickCustomFieldProps(publicData, metadata, 'userType', includeForUserTypes);
// or
// const propsForCustomFields = pickCustomFieldProps(publicData, metadata, 'listingType', includeForListingTypes);

// Inside JSX, we could have more straightforward code:
{propsForCustomFields.map(customFieldProps => {
  const { schemaType, ...fieldProps } = customFieldProps;
  return schemaType === SCHEMA_TYPE_MULTI_ENUM
    ? <SectionMultiEnumMaybe {...fieldProps} />
    : schemaType === SCHEMA_TYPE_TEXT
    ? <SectionMultiEnumMaybe {...fieldProps} />
    : null;
})}
```

Reply:
This is a good idea, however it would require either the listing config and user config having the same name for the "includeForListing|UserType" attribute, or then adding some `(if entitykey == listingType then includeForListingType else includeForUserType)` logic. I think we need to finalise this once we land on some config from Console, i.e. after Mikko's vacation.

Reply:
I moved this section from ProfilePage to util/fieldHelpers.js now, but I didn't yet start working on combining user and listing capabilities i.e. adding it to listing page.

Reply:
@SariSaar Uh, good point! I've been too deep in the filters & EditListingWizard code, that I'm not anymore remembering how the asset data is currently structured.

Mikko might have some opinion about the grouped checkboxes (showConfig&saveConfig), but this structure should not change, as any change there would be inconsistent with the existing listingTypeConfig: 

```
(user|listing)TypeConfig: {
      limitTo(User|listing)TypeIds: true,
      userTypeIds: ['a', 'b', 'c'],
}
```

...and now that I need to add a similar limiter for categories too. It's probably best that I also remove **includeForListingType** property in favor of listingTypeConfig.

Reply:
The helper has now been generalized and should be able to handle both user type and listing type restrictions